voorhoede/head-start
# feature: HTML to Markdown (component & routes)

# Changes

Making our HTML pages available as Markdown is of interest to some people and especially suitable for software such as LLM's. These changes make HTML content available as Markdown.

- Adds a `ToMarkdown` component to render HTML as Markdown (works both run- and build-time).
- Adds a `[locale]/[...path]/index.md.astro` route to render matching route as Markdown (works both run- and build-time).
- Adds post build script to rename Markdown routes to `.md` as Astro always generates `.md/index.html` files.
- Adds support for Github Flavoured Markdown, including tables.

See decision log entry for background.

# Associated issue

N/A

# How to test

1. Open preview link
2. Navigate to a page, like `/en/documentation/getting-started/`
3. Replace the trailing slash with `.md`, like `/en/documentation/getting-started.md`
4. Verify the page is now rendered as Markdown.
5. Navigate to a page with tables, like `/en/demos/tables-demo.md`
6. Verify that tables are rendered to Markdown.

# Checklist
